falcon/request.py

```diff
@@ -1383,7 +1383,7 @@ def get_param_as_bool(self, name, required=False, store=None,
                 the value of the param, but only if the param is found (default
                 ``None``).
             blank_as_true (bool): If ``True``, an empty string value will be
-                treated as ``True`` (default ``False``). Normally empty strings
+                treated as ``True`` (default ``True``). Normally empty strings
```

Comment:
Also, I think it would be helpful to explicitly point out in the docstring that when `blank_as_true` is set to `False` and no value is provided in the query string for the param, an instance of `falcon.HTTPInvalidParam` is raised. Alternatively, see the next suggestion below.

Reply:
What does everyone think about replacing `blank_as_true` with something like `treat_as_flag` or perhaps creating a separate `get_param_as_flag()`? The difference would be that if the param is present and has a blank value, `True` is returned. If the param is missing, instead of raising an error, the method would return `False`.

Reply:
(Addressed in my commits)

Reply:
IMHO the way you proposed in the changeset is sound. Something like ``treat_as_flag``would probably be even better from the theoretical API design point of view, but it would be a yet larger breaking change for Falcon users (unless we supported the old ``blank_as_true`` as well?) for very little benefit.